packages/dd-trace/src/plugins/util/ip_extractor.js

```diff
@@ -41,35 +41,31 @@ function extractIp (config, req) {
   const headers = req.headers
   if (config.clientIpHeader) {
     if (!headers) return
-    const header = headers[config.clientIpHeader]
-    if (!header) return
 
-    return findFirstIp(header)
+    const ip = findFirstIp(headers[config.clientIpHeader])
+    return ip && (ip.public || ip.private)
   }
 
-  const foundHeaders = []
+  let foundPrivateIp
   if (headers) {
     for (let i = 0; i < ipHeaderList.length; i++) {
-      if (headers[ipHeaderList[i]]) {
-        foundHeaders.push(ipHeaderList[i])
+      const firstIp = findFirstIp(headers[ipHeaderList[i]])
+      if (!firstIp) continue
+
+      if (firstIp.public) {
+        return firstIp.public
+      } else if (!foundPrivateIp && firstIp.private) {
+        foundPrivateIp = firstIp.private
       }
     }
   }
 
-  if (foundHeaders.length === 1) {
-    const header = headers[foundHeaders[0]]
-    const firstIp = findFirstIp(header)
-
-    if (firstIp) return firstIp
-  } else if (foundHeaders.length > 1) {
-    log.error(`Cannot find client IP: multiple IP headers detected ${foundHeaders}`)
-    return
-  }
-
-  return req.socket && req.socket.remoteAddress
+  return foundPrivateIp || (req.socket && req.socket.remoteAddress)
 }
 
 function findFirstIp (str) {
+  if (!str) return
```

Comment:
I'd say
```suggestion
  if (!str) return {}
```
So we can replace for example:
`return ip && (ip.public || ip.private)` into `return ip.public || ip.private`

Reply:
Ah nice catch, did not see that